.circleci/config.yml

```diff
@@ -58,29 +97,60 @@ jobs:
 
       - attach_workspace:
           at: .
-
+      - *setup-env
+      - *gcloud-login
```

Comment:
Curious about this steps-reused convention.
Would you like to share the document ?
I only found this blogpost https://circleci.com/blog/persisting-data-in-workflows-when-to-use-caching-artifacts-and-workspaces/ .

Reply:
It is actually a hack on `yaml` file called `anchor` and `aliases`. See [here](https://circleci.com/blog/circleci-hacks-reuse-yaml-in-your-circleci-config-with-yaml/) for reference. I learned it from [hashicorp/consul](https://github.com/hashicorp/consul/blob/master/.circleci/config.yml)

Reply:
Oops...
The above blogpost URL I provided is wrong.
I saw the same article like you share.

Reply:
Just the convention is the same.

I got confused because of 
```
<<: *defaults
```

I thought our codes should be
```
<<: *setup-env
<<: *gcloud-login
```

Reply:
`<<` means [`merge`](https://yaml.org/type/merge.html) the object which the anchor refers to. However, in our case,  we simply reuse it. We can directly use the reference.

Reply:
Thanks for sharing.